recipes-bsp/firmware-imx/firmware-imx_8.6.bb

```diff
@@ -0,0 +1,118 @@
+# Copyright (C) 2012-2016 Freescale Semiconductor
+# Copyright 2017-2019 NXP
+# Copyright (C) 2018 O.S. Systems Software LTDA.
+SUMMARY = "Freescale i.MX firmware"
+DESCRIPTION = "Freescale i.MX firmware such as for the VPU"
+
+require firmware-imx-${PV}.inc
+
+SRC_URI_append = " \
+    file://sdma \
+    file://epdc \
+    file://regulatory \
+    file://hdmi \
+    file://sdma-firmware.service \
+    file://epdc-firmware.service \
+    file://regulatory-firmware.service \
+    file://hdmi-firmware.service \
+"
+
+PE = "1"
+
+inherit allarch systemd
+
+SYSTEMD_PACKAGES = "${PN}-sdma ${PN}-epdc ${PN}-regulatory ${PN}-hdmi"
+SYSTEMD_SERVICE_${PN}-sdma = "sdma-firmware.service"
+SYSTEMD_SERVICE_${PN}-epdc = "epdc-firmware.service"
+SYSTEMD_SERVICE_${PN}-regulatory = "regulatory-firmware.service"
+SYSTEMD_SERVICE_${PN}-hdmi = "hdmi-firmware.service"
+
+do_install() {
+    install -d ${D}${base_libdir}/firmware/imx
+
+    if ${@bb.utils.contains('DISTRO_FEATURES','systemd','true','false',d)}; then
+        install -d ${D}${sysconfdir}
+        install -d ${D}${systemd_system_unitdir}
+        install -m 0755 ${WORKDIR}/sdma ${D}${sysconfdir}
+        install -m 0755 ${WORKDIR}/epdc ${D}${sysconfdir}
+        install -m 0755 ${WORKDIR}/regulatory ${D}${sysconfdir}
+        install -m 0755 ${WORKDIR}/hdmi ${D}${sysconfdir}
+        install -m 0644 ${WORKDIR}/sdma-firmware.service ${D}${systemd_system_unitdir}
+        install -m 0644 ${WORKDIR}/epdc-firmware.service ${D}${systemd_system_unitdir}
+        install -m 0644 ${WORKDIR}/regulatory-firmware.service ${D}${systemd_system_unitdir}
+        install -m 0644 ${WORKDIR}/hdmi-firmware.service ${D}${systemd_system_unitdir}
+    fi
+
+    cd firmware
+    for d in *; do
+        case $d in
+        ddr|seco)
+            # These folders are for i.MX 8 and are included in the boot image via imx-boot
+            bbnote Excluding folder $d
+            ;;
+        *)
+            cp -rfv $d ${D}${base_libdir}/firmware
+            ;;
+        esac
+    done
+    cd -
```

Comment:
Can this whole construct be simplified? For example, there is a mechanism in NXP `imx-gpu-sdk` [recipe](https://source.codeaurora.org/external/imx/meta-imx/tree/meta-sdk/recipes-graphics/imx-gpu-sdk/imx-gpu-sdk_5.4.0.bb?h=zeus-5.4.3-1.0.0#n66) which has a filtering and removal mechanism that is more elegant.